pytorch_lightning/core/hooks.py

```diff
@@ -304,6 +304,12 @@ def configure_sharded_model(self) -> None:
         implementation of this hook is idempotent.
         """
 
+    def on_compress_model(self) -> None:
```

Comment:
Let's rely on the existing hooks. Take an example from https://github.com/tchaton/SparseMLCallback.

It seems the `on_compress_model` could be replaced with `on_fit_end`.

Reply:
@tchaton , thanks for your comments. 
We use on_compress_model hook for two reasons:
1. On_fit_end is for train loop, but static and dynamic quantization didn't need to train, only need to evaluate. 
2. for quantization aware training, we need to tune the quantized model in INC tool, INC will generate quantization configure by specified strategy, and train the model with the lightning trainer, then INC will check the evaluation metric with the lightning trainer, if the metric matches the target, then stop turning，otherwise，will tune again. The whole process is completed in INC tool, so I think the fit processing does not match the requirement.

Reply:
If I understand properly your requirements, you should be able to provide those functionalities by relying on more hooks such as hooks `on_train_end`, `on_validation_end`, and `on_test_end` which could share the code to perform the quantization.

By utilizing the `on_epoch_end` hook should be able to check the validation metrics and decide to stop tuning.

Am I missing something? 

Furthermore, if you believe a hook should be added, please open an issue with a clear explanation on why this is required.

Reply:
Now we only integrate PTQ, so only use on_fit_start hook.